src/hotspot/os/aix/attachListener_aix.cpp

```diff
@@ -419,8 +419,8 @@ bool AttachListener::is_init_trigger() {
   RESTARTABLE(::stat(fn, &st), ret);
   if (ret == -1) {
     log_trace(attach)("Failed to find attach file: %s, trying alternate", fn);
-    snprintf(fn, sizeof(fn), "%s/.attach_pid%d",
-             os::get_temp_directory(), os::current_process_id());
+    os::snprintf_checked(fn, sizeof(fn), "%s/.attach_pid%d",
+                         os::get_temp_directory(), os::current_process_id());
```

Comment:
This could fail if os::get_temp_directory() returns an extremely long path.  How about doing a truncation check like at line 354?

Reply:
Nevermind, same thing.  We would need to fix a lot of code if os::get_temp_directory() returned a pathologically long string.